lib/environment/permissions.js

```diff
@@ -9,52 +9,44 @@ function filterPerms(perms) {
 	return { permissions, origins };
 }
 
-const STORAGE_KEY = 'RES.cachedDeniedPerms';
+export const has = _.memoize(
+	async (...perms) => {
+		const { permissions, origins } = filterPerms(perms);
 
-export function reset() {
-	Storage.delete(STORAGE_KEY);
-}
+		return await sendMessage('permissions', { operation: 'contains', permissions, origins });
+	},
+	(...perms) => perms.join(',')
+);
 
 /**
+ * @param {string} perms Name of site to request permission for
  * @param {...string} perms Optional Chrome permissions to request.
  * @returns {Promise<void, *>} Resolves if the permissions are granted, rejects otherwise.
  */
-export const request = _.memoize(
-	mutex(async (...perms) => {
-		const { permissions, origins } = filterPerms(perms);
-
-		const hasPermissions = await sendMessage('permissions', { operation: 'contains', permissions, origins });
-
-		// already granted, don't ask the user again
-		if (hasPermissions) return;
+export const request = mutex(async (siteName, ...perms) => {
```

Comment:
And if you do that you'll be able to reorder these params; `siteName` could even be optional, perhaps